Jenkinsfile

```diff
@@ -16,84 +16,82 @@ pipeline {
-            }
-        }
-        stage('Release') {
-            when { 
-                branch 'master'
-            }
-            steps {
-                sh 'sh generate-release-w-docker.sh'
-            }
-        }
-        stage('Build final version images') {
-            when {
-                expression {
-                    return isVersionTag(readCurrentTag())
-                }
-            }
-            steps {
-                // TODO: remove build differences based on environments to allow reducing the
-                // number of different images. Maybe `demo` and `production` cold be enought
-                sh '''docker build \
-                    -t "local-doppler-webapp:production-commit-${GIT_COMMIT}" \\
-                    --build-arg environment=production \\
-                    --build-arg react_app_router=browser \\
-                    --build-arg public_url="/" \\
-                    --build-arg version=production-${TAG_NAME}+${GIT_COMMIT} \\
-                    -f Dockerfile.swarm \\
-                    .'''
-                sh '''docker build \
-                    -t "local-doppler-webapp:demo-commit-${GIT_COMMIT}" \\
-                    --build-arg environment=demo \\
-                    --build-arg react_app_router=browser \\
-                    --build-arg public_url="/" \\
-                    --build-arg version=demo-${TAG_NAME}+${GIT_COMMIT} \\
-                    -f Dockerfile.swarm \\
-                    .'''
-                 sh '''docker build \
-                    -t "local-doppler-webapp:int-commit-${GIT_COMMIT}" \\
-                    --build-arg environment=int \\
-                    --build-arg react_app_router=browser \\
-                    --build-arg public_url="/" \\
-                    --build-arg version=int-${TAG_NAME}+${GIT_COMMIT} \\
-                    -f Dockerfile.swarm \\
-                    .'''
-                    sh '''docker build \
-                    -t "local-doppler-webapp:qa-commit-${GIT_COMMIT}" \\
-                    --build-arg environment=qa \\
-                    --build-arg react_app_router=browser \\
-                    --build-arg public_url="/" \\
-                    --build-arg version=qa-${TAG_NAME}+${GIT_COMMIT} \\
-                    -f Dockerfile.swarm \\
-                    .'''
-            }
-        }
         stage('Publish final version images in dopplerdock') {
             environment {
                 DOCKER_CREDENTIALS_ID = "dockerhub_dopplerdock"
                 DOCKER_IMAGE_NAME = "dopplerdock/doppler-webapp"
             }
-            when {
-                expression {
-                    return isVersionTag(readCurrentTag())
+            stages {
+                stage('Publish pre-release images from pull request') {
+                    when {
+                        changeRequest target: 'master'
+                    }
+                    steps {
+                        sh '''docker build \
```

Comment:
![image](https://user-images.githubusercontent.com/1157864/187781510-72bd1a1c-8af0-4ab1-a979-59740168c9c0.png)

Creo que esto no debería llamarse `Publish final version images in dopplerdock` ni `Publish pre-release images from pull request`, en este caso podría ser `Docker Build`, no?